### PR TITLE
Added some debug logging for numpy->swift conversion.

### PR DESCRIPTION
**Pull request by mhong**
e.g. If the numpy array has dtype float64, but we write: `Tensor<Float>(numpy:
numpy)`, the output will be nil.

After turning on debug log, we will see that the correct code is `Tensor<Double>(numpy: numpy)`.

### CODE REVIEW

**Comment by mhong** [comment]
@swift please test tensorflow

**Comment by mhong** [comment]
@swift-ci please test tensorflow

File: stdlib/public/TensorFlow/NumpyConversion.swift [hunk 1]
```diff
@@ -28,16 +28,22 @@ extension ShapedArray : ConvertibleFromNumpyArray
   public init?(numpy numpyArray: PythonObject) {
     // Check if input is a `numpy.ndarray` instance.
     guard Python.isinstance(numpyArray, np.ndarray) == true else {
+      debugLog("numpy array is not an ndarray.")
       return nil
     }
     // Check if the dtype of the `ndarray` is compatible with the `Scalar`
     // type.
     guard Scalar.numpyScalarTypes.contains(numpyArray.dtype) else {
+      debugLog("""
```
Comment (dan-zheng) [inline stdlib/public/TensorFlow/NumpyConversion.swift]:
```swift
      debugLog("""
        'numpy.ndarray' dtype '\(numpyArray.dtype)' cannot be mapped to Swift \
        type '\(Scalar.self)'.
        """)
```

File: stdlib/public/TensorFlow/NumpyConversion.swift [hunk 2]
```diff
@@ -28,16 +28,22 @@ extension ShapedArray : ConvertibleFromNumpyArray
   public init?(numpy numpyArray: PythonObject) {
     // Check if input is a `numpy.ndarray` instance.
     guard Python.isinstance(numpyArray, np.ndarray) == true else {
+      debugLog("numpy array is not an ndarray.")
```
Comment (dan-zheng) [inline stdlib/public/TensorFlow/NumpyConversion.swift]:
```suggestion
      debugLog("PythonObject input has type '\(Python.type(numpyArray)\)' and is not an instance of 'numpy.ndarray'.")
```

File: stdlib/public/TensorFlow/NumpyConversion.swift [hunk 3]
```diff
@@ -28,16 +28,22 @@ extension ShapedArray : ConvertibleFromNumpyArray
...
+      debugLog("""
+                 ndarray dtype \(numpyArray.dtype) cannot be mapped to Swift \
+                 type \(Scalar.self).
+                 """)
       return nil
     }
 
     let pyShape = numpyArray.__array_interface__["shape"]
     guard let shape = Array<Int>(pyShape) else {
+      debugLog("unknown numpy array shape.")
```
Comment (dan-zheng) [inline stdlib/public/TensorFlow/NumpyConversion.swift]:
```suggestion
      debugLog("Cannot access shape of 'numpy.ndarray' instance.")
```

File: stdlib/public/TensorFlow/NumpyConversion.swift [hunk 4]
```diff
@@ -47,6 +53,7 @@ extension ShapedArray : ConvertibleFromNumpyArray
 
     guard let ptrVal =
       UInt(contiguousNumpyArray.__array_interface__["data"].tuple2.0) else {
+      debugLog("cannot get the data pointer of the numpy array.")
```
Comment (dan-zheng) [inline stdlib/public/TensorFlow/NumpyConversion.swift]:
```suggestion
      debugLog("Cannot access data of 'numpy.ndarray' instance.")
```

**Review by dan-zheng — APPROVED** [review]

**Comment by dan-zheng** [comment]
Consider prefixing error messages with `"NumPy conversion error: "`?

**Comment by dan-zheng** [comment]
Updated debug log messages in https://github.com/apple/swift/pull/22317/commits/7a01c24dfc62ebc30ea166245829329df1807c35.

Here's what they look like:
```swift
import TensorFlow
import Python

_RuntimeConfig.printsDebugLog = true

let np = Python.import("numpy")
let npArray = np.ones(PythonObject([2, 2]), dtype: Python.float)
let array = ShapedArray<Int32>(numpy: npArray)! // incompatible dtype
```

```
$ swift numpy.swift
[/Users/danielzheng/swift-dev/swift/stdlib/public/TensorFlow/NumpyConversion.swift:27] NumPy conversion error: 'numpy.ndarray' dtype 'float64' is incompatible with Swift type 'Int32'.
Fatal error: Unexpectedly found nil while unwrapping an Optional value
Stack dump:
...
```

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Review by rxwei — APPROVED** [review]
LGTM. We should do better error messages for Python soon.

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by mhong** [comment]
Thanks for the review and edits.

**Pull request merged**